File: DictionaryProject/WordBuilder/TokenSetCollection.vb

Public Class TokenSetCollection
    Inherits Collections.ObjectModel.Collection(Of TokenSet)

    Public Function FindByName(ByVal name As String) As TokenSet
    	For t As TokenSet In Me
    		If t.Name = name Then
    			Return t
    		End If
    	Next
        Return Nothing
    End Function
End Class

File: monowordbuilder/wordbuilderbase/Project/TokenSetCollection.cs
using System;

namespace Whee.WordBuilder.Project
{
	public class TokenSetCollection : System.Collections.ObjectModel.Collection<TokenSet>
	{
		public TokenSet FindByName(string name)
		{
			foreach (TokenSet t in this) {
				if (t.Name == name) {
					return t;
				}
			}
			return null;
		}

		public int CountByName(string name)
		{
			int result = 0;
			
			foreach (TokenSet t in this) {
				if (t.Name == name) {
					result++;
				}
			}
			return result;
		}
	}
}
